packages/esm-patient-vitals-app/src/vitals/vitals-header/vitals-header.component.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import isEmpty from 'lodash-es/isEmpty';
+import InlineLoading from 'carbon-components-react/es/components/InlineLoading';
+import VitalsHeaderItem from './vitals-header-item.component';
+import VitalsHeaderTitle from './vitals-header-title.component';
+import { useConfig } from '@openmrs/esm-framework';
+import { useVitalsConceptMetadata } from '@openmrs/esm-patient-common-lib';
+import { useVitals } from '../vitals.resource';
+import styles from './vitals-header.component.scss';
+
+interface VitalsHeaderProps {
+  patientUuid: string;
+  showRecordVitals: boolean;
+}
+
+const VitalsHeader: React.FC<VitalsHeaderProps> = ({ patientUuid, showRecordVitals }) => {
+  const config = useConfig();
+  const { t } = useTranslation();
+  const { data: vitals, isLoading } = useVitals(patientUuid);
+  const { data: conceptData } = useVitalsConceptMetadata();
+  const conceptUnits = conceptData?.conceptUnits ? conceptData.conceptUnits : null;
+  const latestVitals = vitals?.[0];
+
+  const [showDetails, setShowDetails] = useState(false);
+
+  const toggleView = () => setShowDetails((prevState) => !prevState);
+
+  return (
+    <>
+      {isLoading ? (
+        <InlineLoading role="progressbar" className={styles.loading} description={`${t('loading', 'Loading')} ...`} />
+      ) : (
+        <div
+          className={`${
+            latestVitals && Object.keys(latestVitals)?.length && !dayjs(latestVitals.date).isToday()
+              ? styles.warningBackground
+              : styles.defaultBackground
+          } ${styles.vitalHeaderStateContainer}`}>
+          <VitalsHeaderTitle
+            showDetails={showDetails}
+            showRecordVitals={showRecordVitals}
+            toggleView={toggleView}
+            view={''}
+            vitals={latestVitals}
+          />
+          {showDetails && (
+            <div>
+              <div className={styles.row}>
+                <VitalsHeaderItem
+                  unitName={t('temperatureAbbreviated', 'Temp')}
+                  unitSymbol={conceptUnits ? conceptUnits[2] : ''}
```

Comment:
At some point (not in this PR) I think the vitals will need some refactoring to make it more robust, especially once we want to make it configurable. e.g. having these really arbitrary indexes tied to specific vitals is not awesome.